## Airline Overbooking

> *CIS 2033, Spring 2017*

>  *April 20, 2017*

### Homework 8, Question 4 

For any given flight, an airline tries to
sell as many tickets as possible. Suppose that on average, 10% of ticket
holders fail to show up, all independent of one another. Knowing this, an
airline will sell more tickets than there are seats available (i.e., overbook
the flight) and hope that there is a sufficient number of ticket holders who
do not show up to compensate for its overbooking. 

Using the Central Limit Theorem, determine $n$, the maximum number of
tickets an airline should sell on a flight with 300 seats so that it can be
approximately 99% confident that all ticket holders who do show up will
be able to board the plane. Use the 1/2-correction we discussed in class in
your calculations.


#### Solution

+ Let $n$ be the number of tickets the airline sells.
+ Let $X_i$ be the Bernoulli random variable indicating whether the $i$-th ticket holder shows up. 
+ We are given that $p$, the probability that the $i$-th ticket holder shows up, is $p = {\bf P} (X_i = 1) = 0.9$.

Then $S_n = X_1 + X_2 + \ldots + X_n$ is a binomial random variable with mean $n \cdot p = 0.9 \cdot n$ and standard deviation of $\sqrt{n \cdot 0.9 \cdot 0.1} = 0.3\sqrt{n}$

We would like
$$
{\bf P}(S_n \leq 300) \approx 0.99
$$

or, using the 1/2 correction,
$$
{\bf P}\Big(\ \frac{S_n - 0.9 \cdot n}{0.3 \cdot \sqrt{n}} \leq \frac{305.5 - 0.9 \cdot n}{0.3 \cdot \sqrt{n}} \  \Big) \approx {\bf P} \Big(Z \leq \frac{305.5 - 0.9 \cdot n}{0.3 \cdot \sqrt{n}} \Big) = 0.99,
$$

where $Z$ stands for the standard normal. Now, the CDF of the standard normal takes the value of $0.99$ when

In [1]:
from scipy.stats import norm
import math

rv = norm()
r = rv.ppf(0.99)       # alternatively, look this value up in the standard normal table
print("The point at which the CDF of the standard normal is 0.99 is: ", r)

The point at which the CDF of the standard normal is 0.99 is:  2.32634787404


So we've got to find $n$ that satisfies the following equation:
$$
\frac{300.5 - 0.9 \cdot n}{0.3 \cdot \sqrt{n}} = 2.3263,
$$

which is the same as 
$$
0.9 \cdot n + 0.3 \cdot 2.3263 \cdot \sqrt{n} - 300.5 = 0
$$

This is a quadratic equation in $\sqrt{n}$, which can be solved using the familiar formula:

In [2]:
a = 0.9
b = 0.3 * 2.32634787404
c = -300.5
solution2quad_eq = (-b + math.sqrt(b**2 - 4*a*c)) / (2 * a)

n = solution2quad_eq**2
print("The number of tickets to sell: ", n)

The number of tickets to sell:  320.01686473120986


Of course you can't sell $320.01686473120986$ tickets, so you go with $\boxed{320}$.